tests/test_infer/test_models/test_baichuan.py

```diff
@@ -0,0 +1,98 @@
+import random
+
+import numpy as np
+import pytest
+import torch
+from model_utils.baichuan2_7B.configuration_baichuan import BaichuanConfig
+from model_utils.baichuan2_7B.modeling_baichuan import BaichuanForCausalLM
+from transformers import AutoTokenizer, GenerationConfig
+
+import colossalai
+from colossalai.inference.config import _DEFAULT_PROMPT_TEMPLATES, InferenceConfig
+from colossalai.inference.core.engine import InferenceEngine
+from colossalai.inference.flash_decoding_utils import FDIntermTensors
+from colossalai.testing import parameterize, rerun_if_address_is_in_use, spawn
+
+
+def setup_seed(seed):
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    np.random.seed(seed)
+    random.seed(seed)
+
+
+def check_inference_engine(use_engine=False, prompt_template=None):
+    setup_seed(20)
+    tokenizer = AutoTokenizer.from_pretrained("baichuan-inc/Baichuan2-7B-Base", use_fast=False, trust_remote_code=True)
+    model = BaichuanForCausalLM(
+        BaichuanConfig(
+            vocab_size=125696, hidden_size=32, intermediate_size=1376, num_attention_heads=1, num_hidden_layers=1
+        )
+    ).cuda()
+    model = model.eval()
+
+    inputs = [
+        "介绍一下今天的北京,比如故宫，天安门，长城或者其他的一些景点,",
+    ]
+
+    output_len = 38
+    do_sample = True
```

Comment:
Why `do_sample` set to `True`? You're not testing on greedy search

Reply:
Done